cmd/webhook/main.go

```diff
@@ -79,32 +79,37 @@ func main() {
 	}
 
 	options := webhook.ControllerOptions{
-		ServiceName:    "tekton-pipelines-webhook",
-		DeploymentName: "tekton-pipelines-webhook",
-		Namespace:      system.GetNamespace(),
-		Port:           8443,
-		SecretName:     "webhook-certs",
-		WebhookName:    "webhook.tekton.dev",
+		ServiceName:                     "tekton-pipelines-webhook",
+		DeploymentName:                  "tekton-pipelines-webhook",
+		Namespace:                       system.GetNamespace(),
+		Port:                            8443,
+		SecretName:                      "webhook-certs",
+		WebhookName:                     "webhook.tekton.dev",
+		ResourceAdmissionControllerPath: "/",
 	}
-	//TODO add validations here
-	controller := webhook.AdmissionController{
-		Client:  kubeClient,
-		Options: options,
-		Handlers: map[schema.GroupVersionKind]webhook.GenericCRD{
-			v1alpha1.SchemeGroupVersion.WithKind("Pipeline"):         &v1alpha1.Pipeline{},
-			v1alpha1.SchemeGroupVersion.WithKind("PipelineResource"): &v1alpha1.PipelineResource{},
-			v1alpha1.SchemeGroupVersion.WithKind("Task"):             &v1alpha1.Task{},
-			v1alpha1.SchemeGroupVersion.WithKind("ClusterTask"):      &v1alpha1.ClusterTask{},
-			v1alpha1.SchemeGroupVersion.WithKind("TaskRun"):          &v1alpha1.TaskRun{},
-			v1alpha1.SchemeGroupVersion.WithKind("PipelineRun"):      &v1alpha1.PipelineRun{},
-			v1alpha1.SchemeGroupVersion.WithKind("Condition"):        &v1alpha1.Condition{},
-		},
-		Logger:                logger,
-		DisallowUnknownFields: true,
-		// Decorate contexts with the current state of the config.
-		WithContext: func(ctx context.Context) context.Context {
-			return v1alpha1.WithDefaultConfigurationName(store.ToContext(ctx))
-		},
+	resourceHandlers := map[schema.GroupVersionKind]webhook.GenericCRD{
+		v1alpha1.SchemeGroupVersion.WithKind("Pipeline"):         &v1alpha1.Pipeline{},
+		v1alpha1.SchemeGroupVersion.WithKind("PipelineResource"): &v1alpha1.PipelineResource{},
+		v1alpha1.SchemeGroupVersion.WithKind("Task"):             &v1alpha1.Task{},
+		v1alpha1.SchemeGroupVersion.WithKind("ClusterTask"):      &v1alpha1.ClusterTask{},
+		v1alpha1.SchemeGroupVersion.WithKind("TaskRun"):          &v1alpha1.TaskRun{},
+		v1alpha1.SchemeGroupVersion.WithKind("PipelineRun"):      &v1alpha1.PipelineRun{},
+		v1alpha1.SchemeGroupVersion.WithKind("Condition"):        &v1alpha1.Condition{},
+	}
+
+	resourceAdmissionController := webhook.NewResourceAdmissionController(resourceHandlers, options, true)
```

Comment:
NIT: The third parameter here is the "DisallowUnknownFields" - but to know that one has to lookup the `NewResourceAdmissionController` code, so I think we lose a bit in readability.

Reply:
@afrittoli I agree :sweat_smile: We should either name that boolean *or* propose some changes upstream to make the `New…` more "functionnal".